iroha/src/account.rs

```diff
@@ -293,25 +263,61 @@ pub mod isi {
     /// #
     /// # let command_payload = &CreateRole {
     /// #    role_name: "user".to_string(),
-    /// #    permissions: Vec::new(),
+    /// #    permissions: vec!["buy".to_string()],
     /// # };
     /// # let result: Vec<u8> = command_payload.into();
     /// let command_payload: CreateRole  = result.into();
     /// ```
     impl std::convert::From<Vec<u8>> for CreateRole {
         fn from(command_payload: Vec<u8>) -> Self {
-            bincode::deserialize(&command_payload).expect("Failed to deserialize payload.")
+            <CreateRole>::decode(&mut command_payload.as_slice())
+                .expect("Failed to deserialize payload.")
         }
     }
 
-    #[test]
-    fn create_role_command_serialization_and_deserialization() {
-        let expected = CreateRole {
-            role_name: "user".to_string(),
-            permissions: Vec::new(),
-        };
-        let actual: CreateRole =
-            bincode::deserialize(&bincode::serialize(&expected).unwrap()[..]).unwrap();
-        assert_eq!(expected, actual);
+    #[cfg(test)]
+    mod tests {
+        use super::*;
+
+        #[test]
+        fn append_role_command_serialization_and_deserialization() {
+            let expected = AppendRole {
+                account_id: Id::new("account", "domain"),
+                role_name: "role".to_string(),
+            };
+            let actual = <AppendRole>::decode(&mut expected.encode().as_slice()).unwrap();
+            assert_eq!(expected, actual);
+        }
+
+        #[test]
+        fn create_account_command_serialization_and_deserialization() {
+            let expected = CreateAccount {
+                account_id: Id::new("account", "domain"),
+                domain_name: "domain".to_string(),
+                public_key: [63; 32],
+            };
+            let actual = <CreateAccount>::decode(&mut expected.encode().as_slice()).unwrap();
+            assert_eq!(expected, actual);
+        }
+
+        #[test]
+        fn add_signatory_command_serialization_and_deserialization() {
+            let expected = AddSignatory {
+                account_id: Id::new("account", "domain"),
+                public_key: [63; 32],
+            };
+            let actual = <AddSignatory>::decode(&mut expected.encode().as_slice()).unwrap();
+            assert_eq!(expected, actual);
+        }
+
+        #[test]
+        fn create_role_command_serialization_and_deserialization() {
+            let expected = CreateRole {
+                role_name: "user".to_string(),
+                permissions: Vec::new(),
+            };
+            let actual = <CreateRole>::decode(&mut expected.encode().as_slice()).unwrap();
+            assert_eq!(expected, actual);
+        }
```

Comment:
All these tests are doing almost the same. Aren't there a way in rust to write a templated test to implement test logic only once?

Reply:
Right now only with addition of external testing framework. Maybe [stainless](https://github.com/reem/stainless) framework is worth considering.

Reply:
But well, latest commit was 11 months ago... Though I haven't seen any other decent non standard testing framework.

Reply:
Also we can use [laze_static](https://docs.rs/lazy_static/1.4.0/lazy_static/) to mimic before all data initialization.

Reply:
me and @eadventurous think about code generation for ISIs - https://soramitsu.atlassian.net/browse/RH2-6